src/python/zquantum/core/wip/circuit/conversions2/cirq_conversions.py

```diff
@@ -1 +1,242 @@
+    (cirq.YPowGate, -0.5): _builtin_gates.RY,
+    (cirq.ZPowGate, -0.5): _builtin_gates.RZ,
+    (cirq.ZPowGate, 0): _builtin_gates.PHASE,
+    (cirq.CZPowGate, 0): _builtin_gates.CPHASE,
+    (cirq.XXPowGate, -0.5): _builtin_gates.XX,
+    (cirq.YYPowGate, -0.5): _builtin_gates.YY,
+    (cirq.ZZPowGate, -0.5): _builtin_gates.ZZ,
+    (cirq.ISwapPowGate, 0.0): _builtin_gates.XY,
+}
+
+CIRQ_GATE_SPECIAL_CASES = {cirq.CSWAP: _builtin_gates.SWAP.controlled(1)}
+
+qubit_index = attrgetter("x")
+
+
+@singledispatch
+def export_to_cirq(obj):
+    """Export given native Zquantum object to its Cirq equivalent.
+
+    This should be primarily used with Circuit objects, but
+    also works for builtin gates and gate operations.
+
+    Exporting of user-defined gates is atm not supported.
+    """
+    raise NotImplementedError(f"{obj} can't be exported to Cirq object.")
+
+
+@export_to_cirq.register
+def export_matrix_factory_gate_to_cirq(gate: _gates.MatrixFactoryGate) -> cirq.Gate:
+    try:
+        cirq_factory = ORQUESTRA_BUILTIN_GATE_NAME_TO_CIRQ_GATE[gate.name]
+        cirq_params = (
+            float(param) if isinstance(param, sympy.Expr) and param.is_Float else param
+            for param in gate.params
+        )
+        return cirq_factory(*cirq_params) if gate.params else cirq_factory
+    except KeyError:
+        raise NotImplementedError(f"Gate {gate} can't be exported to Cirq.")
+
+
+@export_to_cirq.register
+def export_controlled_gate_to_cirq(gate: _gates.ControlledGate) -> cirq.Gate:
+    return export_to_cirq(gate.wrapped_gate).controlled(gate.num_control_qubits)
+
+
+@export_to_cirq.register
+def export_dagger_to_cirq(gate: _gates.Dagger) -> cirq.Gate:
+    return cirq.inverse(export_to_cirq(gate.wrapped_gate))
+
+
+@export_to_cirq.register
+def export_gate_operation_to_cirq(operation: _gates.GateOperation) -> cirq.GateOperation:
+    return export_to_cirq(operation.gate)(*map(cirq.LineQubit, operation.qubit_indices))
+
+
+@export_to_cirq.register
+def export_circuit_to_cirq(circuit: _gates.Circuit) -> cirq.Circuit:
+    return cirq.Circuit([export_to_cirq(operation) for operation in circuit.operations])
+
+
+@singledispatch
+def import_from_cirq(obj):
+    """Import given Cirq object, converting it to its Zquantum native counterpart.
+
+    Currently we support gates corresponding to Zquantum builtin gates, operations
+    on such gates and circuits composed of such gates.
+
+    Also note that only objects using only LineQubits are supported, as currently
+    there is no notion of GridQubit in Zquantum.
+
+    Raises:
+        NotImplementedError: if the object to be imported is of currenlty unsupported type.
+    """
+    try:
+        return CIRQ_GATE_SPECIAL_CASES[obj]
+    except KeyError:
+        raise NotImplementedError(f"{obj} can't be imported into Zquantum.")
+
+
+@import_from_cirq.register
+def convert_eigengate_to_orquestra_gate(eigengate: cirq.EigenGate) -> _gates.Gate:
```

Comment:
(mentioned during online review)
```suggestion
def import_eigengate_to_zquantum_gate(eigengate: cirq.EigenGate) -> _gates.Gate:
```